src/ord/set.rs

```diff
@@ -1024,9 +943,49 @@ where
     }
 }
 
+impl<A, P> DoubleEndedIterator for ConsumingIter<A, P>
+where
+    A: Clone,
+    P: SharedPointerKind,
+{
+    fn next_back(&mut self) -> Option<Self::Item> {
+        self.it.next_back().map(|v| v.0)
+    }
+}
+
+impl<A, P> ExactSizeIterator for ConsumingIter<A, P>
+where
+    A: Clone,
+    P: SharedPointerKind,
+{
+}
+
+impl<A, P> FusedIterator for ConsumingIter<A, P>
+where
+    A: Clone,
+    P: SharedPointerKind,
+{
+}
+
 /// An iterator over the difference between two sets.
 pub struct DiffIter<'a, 'b, A, P: SharedPointerKind> {
-    it: NodeDiffIter<'a, 'b, Value<A>, P>,
+    it: map::DiffIter<'a, 'b, A, (), P>,
+}
+
+/// A description of a difference between two ordered maps.
+#[derive(PartialEq, Eq, Debug)]
+pub enum DiffItem<'a, 'b, A> {
+    /// This value has been added to the new map.
+    Add(&'b A),
+    /// This value has been changed between the two maps.
+    Update {
```

Comment:
I realize this isn't really new code, but I'm still confused: shouldn't this case never happen for sets? We require the key type to impl `Ord`, and `Ord` requires that if neither key is bigger than the other then they're equal. And the value type for sets is `()`, which is always equal to every other `()`. So I think you should only ever see `Add` or `Remove`, right?

Reply:
Indeed, let me remove it